cmake/OpenCVGenConfig.cmake

```diff
@@ -128,8 +128,11 @@ endif()
 #  Part 3/3: ${BIN_DIR}/win-install/OpenCVConfig.cmake  -> For use within binary installers/packages
 # --------------------------------------------------------------------------------------------
 if(WIN32)
-  set(OpenCV_INCLUDE_DIRS_CONFIGCMAKE "\"\${OpenCV_CONFIG_PATH}/include\" \"\${OpenCV_CONFIG_PATH}/include/opencv\"")
-  set(OpenCV2_INCLUDE_DIRS_CONFIGCMAKE "\"\"")
+
+  if (CMAKE_HOST_SYSTEM_NAME MATCHES Windows)
+    set(OpenCV_INCLUDE_DIRS_CONFIGCMAKE "\"\${OpenCV_CONFIG_PATH}/include\" \"\${OpenCV_CONFIG_PATH}/include/opencv\"")
+    set(OpenCV2_INCLUDE_DIRS_CONFIGCMAKE "\"\"")
+  endif ()
```

Comment:
@alalek, the target os is alway win32,
I differentiate the case when the host os is linux (cross-compiling), in which case the values are taken from previous values as you mention, and the case when the host is windows (we go inside that if).